sdk/spring/azure-spring-boot-core/src/main/java/com/azure/spring/core/AzureProperties.java

```diff
@@ -0,0 +1,362 @@
+
+import javax.annotation.PostConstruct;
+
+/**
+ * Azure related properties.
+ */
+@Validated
+@ConfigurationProperties(AzureProperties.PREFIX)
+public class AzureProperties {
+
+    public static final String PREFIX = "spring.cloud.azure";
+
+    /**
+     * Client id to use when performing service principal authentication with Azure.
+     */
+    private String clientId;
+
+    /**
+     * Client secret to use when performing service principal authentication with Azure.
+     */
+    private String clientSecret;
+
+    /**
+     * Tenant id for the Azure resources.
+     */
+    private String tenantId;
+
+    /**
+     * Name of the Azure resource group.
+     */
+    private String resourceGroup;
+
+    /**
+     * Name of the Azure cloud to connect to.
+     */
+    private AzureCloud environment = AzureCloud.Azure;
+
+    /**
+     * The Azure Active Directory endpoint to connect to.
+     */
+    private String authorityHost;
+
+    /**
+     * Name of the region where resources would be automatically created.
+     */
+    private String region;
+
+    /**
+     * Flag to automatically create resources.
+     */
+    private boolean autoCreateResources = false;
+
+    /**
+     * Flag to enable MSI.
+     */
+    private boolean msiEnabled = false;
+
+    /**
+     * Subscription id to use when connecting to Azure resources.
+     */
+    private String subscriptionId;
+
+    /**
+     * URL of the proxy for HTTP connections.
+     */
+    private String httpProxy;
+
+    /**
+     * URL of the proxy for HTTPS connections.
+     */
+    private String httpsProxy;
+
+    /**
+     * Endpoint to connect to when using Azure Active Directory managed service identity (MSI).
+     */
+    private String identityEndpoint;
+
+    /**
+     * Header when connecting to Azure Active Directory using managed service identity (MSI).
+     */
+    private String identityHeader;
+
+    /**
+     * A list of hosts or CIDR to not use proxy HTTP/HTTPS connections through, separated by comma.
+     */
+    private String noProxy;
+
+    /**
+     * Endpoint to connect to when using Azure Active Directory managed service identity (MSI).
+     */
+    private String msiEndpoint;
+
+    /**
+     * Secret when connecting to Azure Active Directory using managed service identity (MSI).
+     */
+    private String msiSecret;
+
+    /**
+     * Username to use when performing username/password authentication with Azure.
+     */
+    private String username;
+
+    /**
+     * Password to use when performing username/password authentication with Azure.
+     */
+    private String password;
+
+    /**
+     * Path of a PEM certificate file to use when performing service principal authentication with Azure.
+     */
+    private String clientCertificatePath;
+
+    /**
+     * Flag to disable the CP1 client capabilities in Azure Identity Token credentials.
+     */
+    private boolean identityDisableCP1;
```

Comment:
What is CP1?

Reply:
It is used for [claim challenge](https://docs.microsoft.com/en-us/azure/active-directory/develop/claims-challenge#client-capabilities) while currently not used in our library.